<launch>

  <!-- Assign home directory -->
  <arg name="home_dir" default="/home/shuai"/>

  <!-- Choose scenario -->
  <arg name="scenario" default="0011"/>

  <!-- Choose replay speed -->
  <arg name="speed" default="0.65"/>

  <!-- Choose replay delay -->
  <arg name="delay" default="1" />

  <!-- Play rosbag file -->
  <node pkg="rosbag" type="play" name="player" args="-r $(arg speed) -d $(arg delay) $(arg home_dir)/master_ws/kitti_data/$(arg scenario)/synchronized_data.bag"/>
  <!-- <node pkg="rosbag" type="play" name="player" args="-r $(arg speed) -d $(arg delay) $(arg home_dir)/master_ws/kitti_data/$(arg scenario)/kitti_2011_09_28_drive_0047_synced.bag"/> -->

  <!-- Run the sensor setup node -->
  <node pkg="roscuda_lidar_ogm" type="sensor_setup" name="sensor_setup_node" output="screen">

    <!-- Get parameters from parameter.yaml file -->
    <rosparam file="$(find roscuda_lidar_ogm)/config/parameters.yaml" command="load" />
	  
    <!-- Get data path -->
    <param name="home_dir" value="$(arg home_dir)" />

    <!-- Get scenario identifier -->
    <param name="scenario" value="$(arg scenario)" />
  </node>

  <!-- Launch rviz for visualization -->
  <node pkg="rviz" type="rviz" name="my_rviz"
	args="-d $(find roscuda_lidar_ogm)/config/rviz_config_sensor.rviz"/>
  
</launch>
